### PR TITLE
feat: bump snaps packages and support `InputChangeEvent` for dynamic UI

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Bump snaps packages to latest, primarily to support `InputChangeEvent`. Also includes changes to the dynamic UI functionality to enable `InputChangeEvent`.

`InputChangeEvent` will fire 300ms after the last change to any input form input and trigger the `onUserInput` export in the Snap. This PR also changes the way keys are generated for each section of the Snap UI. The previous logic for generating keys was replaced with a hashing scheme that includes more values from the component in order to be more resistant to duplicates without constantly re-rendering.

Summary of changes in the snaps deps:
- Improve timeout handling when the execution environment fails to load
- Bump LavaMoat
- Add `InputChangeEvent`
- Add `error` prop to `input` component

### CODE REVIEW

**Comment by FrederikBolding** [comment]
@metamaskbot update-policies

File: ui/components/app/snaps/snap-ui-renderer/utils.ts [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
 
 export type MapToTemplateParams = {
+  map: Record<string, number>;
   element: Component;
-  elementKeyIndex: { value: number };
-  rootKey: string;
   form?: string;
 };
 
+const generateHash = memoize((component: Component) => {
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-renderer/utils.ts:13]: Can you leave a comment explaining what this does ?

File: ui/components/app/snaps/snap-ui-renderer/utils.ts [hunk 2]
```diff
@@ -1,17 +1,44 @@
...
+          type,
+          name: name ?? null,
+          value,
+        }),
+      ),
+    ),
+  );
+});
+
+function generateKey(map: Record<string, number>, component: Component) {
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-renderer/utils.ts:32]: Same here

File: ui/contexts/snaps/snap-interface.tsx [hunk 3]
```diff
@@ -16,7 +16,12 @@ import { getMemoizedInterface } from '../../selectors';
 import { handleSnapRequest, updateInterfaceState } from '../../store/actions';
 import { mergeValue } from './utils';
 
-export type HandleEvent = (event: UserInputEventType, name?: string) => void;
+export type HandleEvent = (
+  event: UserInputEventType,
+  name?: string,
+  value?: string,
+  flush?: boolean,
+) => void;
```
Comment (GuillaumeRx) [inline ui/contexts/snaps/snap-interface.tsx:24]: We could maybe use an object here rather than multiple optional params

**Review by david0xd — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**